src/imports.rs

```diff
@@ -1126,18 +1126,107 @@ impl Rewrite for UseTree {
 
     // This does NOT format attributes and visibility or add a trailing `;`.
     fn rewrite_result(&self, context: &RewriteContext<'_>, mut shape: Shape) -> RewriteResult {
-        let mut result = String::with_capacity(256);
-        let mut iter = self.path.iter().peekable();
-        while let Some(segment) = iter.next() {
-            let segment_str = segment.rewrite_result(context, shape)?;
-            result.push_str(&segment_str);
-            if iter.peek().is_some() {
-                result.push_str("::");
-                // 2 = "::"
-                shape = shape.offset_left(2 + segment_str.len(), self.span())?;
+        if context.config.style_edition() >= StyleEdition::Edition2024 {
+            fn proceed(
+                context: &RewriteContext<'_>,
+                span: &Span,
+                shape: &Shape,
+                curr_segment: &UseSegment,
+                curr_segment_is_allow_overflow: bool,
+                next_segment: Option<&&UseSegment>,
+            ) -> Result<(String, Shape), RewriteError> {
+                let mut rewritten_segment = curr_segment.rewrite_result(context, shape.clone())?;
+                if next_segment.is_some() {
+                    rewritten_segment.push_str("::");
+                }
+                let reserved_room_for_brace = match next_segment.map(|s| &s.kind) {
+                    Some(UseSegmentKind::List(_)) => "{".len(),
+                    _ => 0,
+                };
```

Comment:
It would be helpful to add a comment explaining why `reserved_room_for_brace` is calculated?

Reply:
Added. Thanks!